langchain/document_loaders/blockchain.py

```diff
@@ -55,34 +61,77 @@ def __init__(
             raise ValueError(f"Invalid contract address {self.contract_address}")
 
     def load(self) -> List[Document]:
-        url = (
-            f"https://{self.blockchainType}.g.alchemy.com/nft/v2/"
-            f"{self.api_key}/getNFTsForCollection?withMetadata="
-            f"True&contractAddress={self.contract_address}"
-            f"&startToken={self.startToken}"
-        )
+        result = []
+
+        current_start_token = self.startToken
+
+        while True:
```

Comment:
Open to the concept- but want to make sure I understand your thinking.  Are you suggesting we exit the loop if the set time is exceeded, or only add the time check for testing purposes?

Reply:
Yeah I think it would be convenient to provide users a way to early stop based on time check. This could help us mitigate very extreme cases when the expected time cost is very high.

A similar idea is in use in `AgentExecutor` where a single user request may requires multiple rounds of immediate searching. Code pointer below. I think it’s a good idea. Maybe we can consider something similar here?

https://github.com/hwchase17/langchain/blob/master/langchain/agents/agent.py#L830

Reply:
@skcoirz I took a look at the asyncio process and impelemented a general asyncio-based approach in a new blockchain_async file that I intend to remove from my next commit but keeping here for review.  This approach must have a glitch as it still takes 45 seconds to load, or it reaches api rate limits on the api.

I updated blockchain.py doc loader with a simple check against a new max_execution_time param that will exit the flow if this number is exceeded.  I believe this is a simple approach that can accomplish the goal, but happy to continue trying the asyncio model as well.